Sources/VLCAppDelegate.m

```diff
@@ -353,7 +343,25 @@ - (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDiction
                     errorCallback = [NSURL URLWithString:value];
             }
             if ([action isEqualToString:@"stream"] && movieURL) {
-                [self playWithURL:movieURL successCallback:successCallback errorCallback:errorCallback];
+                [self playWithURL:movieURL completion:^(BOOL success) {
+                    if (@available(iOS 10, *)) {
+                        if (success) {
+                            [[UIApplication sharedApplication] openURL:successCallback options:@{} completionHandler:nil];
+                        } else {
+                            [[UIApplication sharedApplication] openURL:errorCallback options:@{} completionHandler:nil];
+                        }
+                    } else {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+                        /* UIApplication's replacement calls require iOS 10 or later, which we can't enforce as of yet */
+                        if (success) {
+                            [[UIApplication sharedApplication] openURL:successCallback];
+                        } else {
+                            [[UIApplication sharedApplication] openURL:errorCallback];
+                        }
```

Comment:
How would you feel about simplifying the success checks by caching the callback?
Something like:

```
NSURL *callback = success ? successCallback : errorCallback;
if (@available(iOS 10, *)) {
    [[UIApplication sharedApplication] openURL:callback options:@{} completionHandler:nil];
} else {
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wdeprecated-declarations"
    /* UIApplication's replacement calls require iOS 10 or later, which we can't enforce as of yet */
    [[UIApplication sharedApplication] openURL:callback];
#pragma clang diagnostic pop
```

Reply:
yes definitely!